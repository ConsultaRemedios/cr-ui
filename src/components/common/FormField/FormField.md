### Basic Usage
FormField is a component responsible to group label, field and error messages
```js
  <FormField label="Name">
    <BaseInput placeholder="type your name"/>
  </FormField>
```

### FormField with error
```js
  <FormField label="Name"
    error="campo obrigatÃ³rio"
  >
    <BaseInput placeholder="type your name" :has-error="true"/>
  </FormField>
```

### FormField with focus
```js
  <FormField label="Name">
    <BaseInput placeholder="type your name" :autofocus="true" />
  </FormField>
```


### FormField without label
```js
  <FormField>
    <BaseInput placeholder="type your name" />
  </FormField>
```
