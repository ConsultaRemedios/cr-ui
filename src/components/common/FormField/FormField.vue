<template>
  <div :data-form-field-with-error="!!error">
    <div v-if="label">
      <BaseLabel
        :is-for="isFor"
        :has-error="!!error"
        :required="required"
        :disabled="disabled"
      >
        {{ label }}

        <template slot="outerSlot">
          <div :class="$style.innerSlot">
            <slot></slot>
          </div>
        </template>
      </BaseLabel>
    </div>

    <slot v-if="!label" />

    <div
      v-if="error"
      :class="$style.errorMessage"
    >
      <span>
        {{ error | capitalize }}
      </span>
    </div>
  </div>
</template>

<script>
import BaseLabel from '../BaseLabel';
import { capitalize } from '../../../filters';

export default {
  name: 'FormField',
  components: {
    BaseLabel,
  },
  filters: {
    capitalize,
  },
  props: {
    label: {
      type: String,
      default: '',
    },
    isFor: {
      type: String,
    },
    error: {
      type: String,
      default: '',
    },
    required: {
      type: Boolean,
      default: true,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style module>
  .errorMessage {
    display: block;
    text-align: left;
    margin-top: 10px;
    color: #D30B3F;
  }

  .innerSlot {
    margin-top: 8px;
  }
</style>
