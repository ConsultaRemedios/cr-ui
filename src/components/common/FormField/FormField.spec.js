import { shallowMount } from '@vue/test-utils';
import FormField from './FormField.vue';

const mountComponent = (props) => {
  const propsData = props ? props : {};
  return shallowMount(FormField, { propsData });
};

describe('Form Field component', () => {
  it('matches snapshot', () => {
    const wrapper = mountComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('When there is label', () => {
    it('matches snapshot', () => {
      const wrapper = mountComponent({ label: 'Name' });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('When there is error', () => {
    it('matches snapshot', () => {
      const wrapper = mountComponent({ label: 'Name', error: 'a name error' });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
