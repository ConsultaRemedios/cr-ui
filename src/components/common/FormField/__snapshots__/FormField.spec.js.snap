// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Field component When there is error matches snapshot 1`] = `
<div
  data-form-field-with-error="true"
>
  <div>
    <baselabel-stub
      haserror="true"
      required="true"
    >
      
      Name

      
      <template>
        <div
          class="innerSlot"
        />
      </template>
    </baselabel-stub>
  </div>
   
  <!---->
   
  <div
    class="errorMessage"
  >
    <span>
      
      A name error
    
    </span>
  </div>
</div>
`;

exports[`Form Field component When there is label matches snapshot 1`] = `
<div>
  <div>
    <baselabel-stub
      required="true"
    >
      
      Name

      
      <template>
        <div
          class="innerSlot"
        />
      </template>
    </baselabel-stub>
  </div>
   
  <!---->
   
  <!---->
</div>
`;

exports[`Form Field component matches snapshot 1`] = `
<div>
  <!---->
    
  <!---->
</div>
`;
